src/raftstore/store/peer_storage.rs

```diff
@@ -1289,12 +1289,14 @@ pub fn do_snapshot(
     let mut snap_data = RaftSnapshotData::new();
     snap_data.set_region(state.get_region().clone());
     let mut stat = SnapshotStatistics::new();
+    let mut copy = mgr.clone();
     s.build(
         snap,
         state.get_region(),
         &mut snap_data,
         &mut stat,
         Box::new(mgr.clone()),
+        &mut copy.get_limiter(),
```

Comment:
Use Arc<RateLimiter> directly, don't use mutable here.

Btw, clone the RateLimiter, not mgr.

Reply:
I am working on it.